File: backend/src/main/java/cohappy/backend/model/dto/response/UserAccountDTO.java
package cohappy.backend.model.dto.response;

import lombok.AllArgsConstructor;
import lombok.Data;
import org.springframework.data.annotation.Id;

import java.time.LocalDate;
import java.util.List;

@Data
@AllArgsConstructor
public class UserAccountDTO{
    private String name;
    private String surname;
    private LocalDate birthDate;
    private int age;
    @Id
    private String userCode;
    private List<byte[]> images;
    private String email;
    private String phoneNumber;
    private String password;
    private PortfolioDTO portfolio;
}

File: frontend/app/src/main/java/cohappy/frontend/model/dto/response/UserAccountDTO.kt
package cohappy.frontend.model.dto.response

import java.time.LocalDate
import com.squareup.moshi.JsonClass
@JsonClass(generateAdapter = true)
data class UserAccountDTO(
    var name: String? = null,
    var surname: String? = null,
    var birthDate: String? = null,
    var cv: String? = null,
    var age: Int? = null,
    var userCode: String? = null, // Mettiamo il nullable per sicurezza
    var images: List<String>? = null, // Da ByteArray a String.
    var email: String? = null,
    var phoneNumber: String? = null,
    var password: String? = null,
    var portfolio: PortfolioDTO? = null
)